Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -103,58 +121,40 @@ public class ApolloCodegen {
     try validate(configContext, with: compilationResult)
 
     let ir = IR(compilationResult: compilationResult)
+    
+    if buildOptions.contains(.operationManifest) {
+      var operationIDsFileGenerator = OperationManifestFileGenerator(config: configContext)
+      
+      for operation in compilationResult.operations {
+        let irOperation = ir.build(operation: operation)
```

Comment:
`ir.build(operation:)` does a _LOT_ of computing and each operation hold a lot in memory (thats why they are in an `autoreleasepool` in the code generation below. We should be careful to only do this once per operation.

But looking at the `OperationManifestFileGenerator`, we actually don't need the built operation to do this. We should be able to change `collectOperationIdentifier()` to take in the `CompilationResult.OperationDefinition` instead of the `IR. Operation`.

Reply:
Looks like we do need the `operationIdentifier` and `referencedFragments` which are on the `IR.Operation` and not on the `CompilationResult.OperationDefinition`. Is there something I am missing here since it looks like those are calculated as part of the IR process?

Reply:
Oooof you're right. Okay well, then in that case, we should try to build the operations once and use it for both things. We also need to make sure that we _release_ the operation after we're done using it, too much memory usage on those objects.